annnufan/assignment1/solution_1.cpp

```diff
@@ -0,0 +1,118 @@
+#include <iostream>
+#include <string>
+#include <cstring>
+#include <vector>
+#include <algorithm>
+#include <cctype>
+#include <functional>
+
+bool word_anagram(std::string word_1, std::string word_2, bool case_sens) {
+	if (!case_sens) {
+		std::transform(word_1.begin(), word_1.end(), word_1.begin(), ::tolower);
+		std::transform(word_2.begin(), word_2.end(), word_2.begin(), ::tolower);
+	}
+	if (word_1.length() != word_2.length()) {
+		return false;
+	}
+
+	std::vector<int> count(256, 0);

```

Comment:
The number range of `char` type is -127 to 128.
So you can use an `unsigned char` instead (which range is 0-255).
Another option is to assume for the scope of this assignment that "extended char" values are not used (i.e. only 0-127 values are possible). In that case, `count` vector can be of size 128 instead.
(Either way, signed char can never hold value of 128 or more).

Reply:
Could I add 127 for all  char before counting? I think it's be solve of problem

Reply:
You can avoid the +127 in this case: You can use the implicit conversion from `char` to `unsigned char` to do the work for you. It is also more safe (an unsigned char can never be negative!).

So you can explicitly state the type in the loop:
```C++
for (unsigned char ch : word_1) { ... }
```
This way each character will be converted to 0-255 range and accessing the array is safe.